src/tribler/core/components/torrent_checker/torrent_checker/dataclasses.py

```diff
@@ -15,13 +16,23 @@
 HEALTH_FRESHNESS_SECONDS = 4 * HOUR  # Number of seconds before a torrent health is considered stale. Default: 4 hours
 
 
+class Source(IntEnum):
+    """ Source of the Torrent Health information.
+    """
+    NONE = 0
+    TRACKER = 1
+    DHT = 2
```

Comment:
I think this enum should not include `TRACKER` and `DHT` items. They represent orthogonal information. The source enum values can be something like `Uknown`, `Local`, `OldPopularityCommunity`, `PopularityCommunityRandomTorrent`, or `PopularityCommunityPopularTorrent`, and we can extend this list in the future.

Reply:
Added, but we need more @xoriole input here.

Reply:
I agree with the current changes: NONE, TRACKER, DHT, TRACKER, POPULARITY_COMMUNITY

Reply:
@xoriole let me explain my thoughts in more detail.

Right now, when PopularityCommunity gossips HealthInfo, it does not pass information about the origin of that info - whether it is obtained from DHT or a tracker. Sometimes, the gossiped health contains a surprising number of seeders that do not correspond with the results of local checks.

One possible explanation is that the results of local checks and the obtained from PopularityCommunity were obtained from different sources. For example, local torrent metadata for a specific torrent hash can include trackers A and B, both dead. At the same time, on a different peer, the same torrent hash can be associated with a different metainfo that includes a tracker C with many seeders for that torrent.

For that reason, it may be beneficial in a future version of PopularityCommunity to gossip about the source of the information, including the tracker URL. It can help to understand the difference in results of checks and also can allow spreading the information about the actual live trackers.

Also, it may be beneficial to know WHY PopularityCommunity decided to gossip about some health information: was it because the torrent was randomly selected, because it is popular, or for some other reason. It allows us to see the proportion of torrents gossiped for different reasons and experiment with various algorithms for choosing which torrent to gossip about.

For that reason, it may be beneficial to separate three orthogonal types of information in the HealthInfo structure:
* **How** the torrent health was calculated - via DHT, BEP33, or some specific tracker
* **Who** calculated the torrent health - the local instance or a remote peer
* **Why** the health was sent to us - as a random torrent, as a popular torrent, or for a different reason

Currently, we do not store all this information in the database. But it does not harm to add these fields now when we create a new database structure, as it may be harder to change the database structure in the future. It does not require changes in the existing PopularityCommunity, as it can use just a subset of possible combinations of fields.

So, I don't think we should mix enum items DHT/TRACKER/POPULARITY_COMMUNITY in the same field. Yes, right now, we don't know whether the information gossiped by the popularity community was obtained from DHT or from some tracker. But in the future, we can extend the format of the gossip message and include this information, and in my opinion, it is better to put orthogonal information into different fields from the beginning.

Reply:
So, mixing DHT / TRACKER / POPULARITY_COMMUNITY should work now but can restrict us later. On the other side, keeping them in separate fields will allow us to send tracker info via an extended version of PopularityCommunity, and it can be beneficial for us while being fully backward-compatible and supporting the current protocol of PopularityCommunity.

Reply:
In my opinion, source as DHT / TRACKER / POPULARITY_COMMUNITY is sufficient for now.

While knowing tracker details could be useful we cannot trust this information as received via Popularity Community. I would consider only health info obtained from self-check to be somewhat reliable (not entirely). So, I do not see a clear picture of how having tracker info could be useful.

Regarding the questions:
- **How the torrent health was calculated - via DHT, BEP33, or some specific tracker:** This is specified by the external peer so might not be reliable what value it has.
- **Who calculated the torrent health - the local instance or a remote peer:** If the source is DHT or Tracker, I'd consider it as local. If it is Popularity community then it is remote.
- **Why the health was sent to us - as a random torrent, as a popular torrent, or for a different reason** - Since only health checked torrent info is shared, there might not be much difference here.